odoo/osv/expression.py

```diff
@@ -893,7 +894,8 @@ def child_of_domain(left, ids, left_model, parent=None, prefix=''):
                     records = records.search([(parent_name, 'in', records.ids)], order='id') - records.browse(child_ids)
                 domain = [('id', 'in', list(child_ids))]
             if prefix:
-                return [(left, 'in', left_model_sudo._search(domain))]
+                rhs = list(child_ids) or left_model_sudo._search(domain)
+                return [(left, 'in', rhs)]
```

Comment:
IMO, it will be more understandable to duplicate `if prefix:` for each branch of the previous `if left_model._parent_store`, no ?

Reply:
Maybe for the readability. But duplicating the `if` generates a better solution if `child_ids` is a empty list in the second case (`else` clause, where ids is also empty).